src/poll.rs

```diff
@@ -76,106 +76,113 @@ impl PollWatcher {
-            // TODO: handle metadata events
-            // TODO: handle renames
-            // TODO: DRY it up
+        let _ = thread::Builder::new()
+            .name("notify-rs poll".to_string())
+            .spawn(move || {
+                // In order of priority:
+                // TODO: handle metadata events
+                // TODO: handle renames
+                // TODO: DRY it up
 
-            loop {
-                if !open.load(Ordering::SeqCst) {
-                    break;
-                }
+                loop {
+                    if !open.load(Ordering::SeqCst) {
+                        break;
+                    }
 
-                if let Ok(mut watches) = watches.lock() {
-                    let current_time = Instant::now();
+                    if let Ok(mut watches) = watches.lock() {
+                        let current_time = Instant::now();
 
-                    for (
-                        watch,
-                        &mut WatchData {
-                            is_recursive,
-                            ref mut paths,
-                        },
-                    ) in watches.iter_mut()
-                    {
-                        match fs::metadata(watch) {
-                            Err(e) => {
-                                let err = Err(Error::io(e).add_path(watch.clone()));
-                                event_handler(err);
-                                continue;
-                            }
-                            Ok(metadata) => {
-                                if !metadata.is_dir() {
-                                    let mtime =
-                                        FileTime::from_last_modification_time(&metadata).seconds();
-                                    match paths.insert(
-                                        watch.clone(),
-                                        PathData {
-                                            mtime,
-                                            last_check: current_time,
-                                        },
-                                    ) {
-                                        None => {
-                                            unreachable!();
-                                        }
-                                        Some(PathData {
-                                            mtime: old_mtime, ..
-                                        }) => {
-                                            if mtime > old_mtime {
-                                                let kind = MetadataKind::WriteTime;
-                                                let meta = ModifyKind::Metadata(kind);
-                                                let kind = EventKind::Modify(meta);
-                                                let ev = Event::new(kind).add_path(watch.clone());
-                                                event_handler(Ok(ev));
+                        for (
```

Comment:
is all this indentation noise from rustfmt ?

Reply:
Yeah - it just jumps in a bit because the scope block indents one further.